src/runtime_src/core/edge/common/aie_parser.cpp

```diff
@@ -208,6 +209,42 @@ get_tiles(const pt::ptree& aie_meta, const std::string& graph_name)
   return tiles;
 }
 
+std::vector<tile_type>
+get_event_tiles(const pt::ptree& aie_meta, const std::string& graph_name,
+                e_tile_type type)
+{
+  std::vector<tile_type> tiles;
+
+  // Not supported yet
+  if (type == xrt_core::edge::aie::AIE_TILE_SHIM)
+    return tiles;
+
+  const char* key_cols[xrt_core::edge::aie::AIE_TILE_NUM_TYPES] =
+      {"core_columns", "dma_columns", "N/A"};
+  const char* key_rows[xrt_core::edge::aie::AIE_TILE_NUM_TYPES] =
+      {"core_rows", "dma_rows", "N/A"};
```

Comment:
Regarding the enum see later.

`const std::array<const char*,  static_cast<int>(xrt_core::edge::aie::e_tile_type::tile_num_types)> key_cols = { ... };`

Not much advantage, but clang_tidy would object to the old-style arrays.